src/Data/Array/Accelerate/Smart.hs

```diff
@@ -1190,6 +1191,11 @@ mkVectorIndex = let n :: Int
                     n = fromIntegral $ natVal $ Proxy @n
                 in mkPrimBinary $ PrimVectorIndex @n (VectorType n singleType) integralType
 
+mkVectorWrite :: forall n a. (KnownNat n, VecElt a) => Exp (Vec n a) -> Exp Int -> Exp a -> Exp (Vec n a)
+mkVectorWrite = let n :: Int
+                    n = fromIntegral $ natVal $ Proxy @n
+                in mkPrimTernary $ PrimVectorWrite @n (VectorType n singleType) integralType
```

Comment:
Is there any (`-ddebug` mode) bounds checking for this? Or, what happens when that `Exp Int` index exceeds the length of the `Vec n a`?

Reply:
good idea, is there a way to add it only in debug mode?

Reply:
Ah apparently bounds checks are behind a separate compile flag, and not `-ddebug` (which only adds debug symbols). So you can use [`boundsChecksAreEnabled`](https://github.com/robbert-vdh/accelerate/blob/58a88ceccb3106ec24179812f45eb8461f9755e8/src/Data/Array/Accelerate/Debug/Internal.hs#L45-L51) to check if they are enabled, and you need to configure Accelerate with the `+bounds-checks` flag to enable them.